core/src/main/java/org/kiwix/kiwixmobile/core/utils/files/FileUtils.kt

```diff
@@ -219,4 +220,34 @@ object FileUtils {
       File("$fileName.part").exists() -> "$fileName.part"
       else -> "${fileName}aa"
     }
+
+  @JvmStatic fun writeLogFile() {
+    if (isExternalStorageWritable) {
+      val appDirectory =
+        File(Environment.getExternalStorageDirectory().toString() + "/Kiwix")
+      val logFile = File(appDirectory, "logcat.txt")
+      Log.d("KIWIX", "Writing all logs into [" + logFile.path + "]")
+
+      // create a new app folder
+      if (!appDirectory.exists()) {
+        appDirectory.mkdir()
+      }
+
+      if (logFile.exists() && logFile.isFile) {
+        logFile.delete()
+      }
+      // clear the previous logcat and then write the new one to the file
+      try {
+        logFile.createNewFile()
+        getRuntime().exec("logcat -c")
+        Runtime.getRuntime().exec("logcat -f " + logFile.path + " -s kiwix")
+      } catch (e: IOException) {
+        Log.e("KIWIX", "Error while writing logcat.txt", e)
```

Comment:
And as we are extensively using the logging technique, throughout the codebase, then how would it be if we make use of some good logging library like _Timber_?

Reply:
https://github.com/kiwix/kiwix-android/issues/1270 an issue exists, hasn't been high priority though